File: backend/src/main/java/com/whatcanicook/dto/model/RecipeSummaryDto.java
package com.whatcanicook.dto.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class RecipeSummaryDto {

    private Long id;

    private String title;

    private String description;

    private boolean publicRecipe;

    private Long userId;

    private String username;

    private LocalDateTime createdAt;

    private Integer matchingIngredients;

    private Integer totalIngredients;
}

File: frontend/app/src/main/java/com/app/dto/model/RecipeSummaryDto.kt
package com.app.dto.model

data class RecipeSummaryDto(

    val id: Long,
    val title: String,
    val description: String,
    val publicRecipe: Boolean,
    val userId: Long,
    val username: String,
    val createdAt: String,
    val matchingIngredients: Int? = null,
    val totalIngredients: Int? = null
)
